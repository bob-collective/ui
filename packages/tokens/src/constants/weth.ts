import { ChainId } from '@gobob/chains';
import { ERC20Token } from '@gobob/currency';

export const WETH9 = {
  [ChainId.ETHEREUM]: new ERC20Token(
    ChainId.ETHEREUM,
    '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
    18,
    'WETH',
    'Wrapped Ether',
    'https://weth.io'
  ),
  [ChainId.SEPOLIA]: new ERC20Token(
    ChainId.SEPOLIA,
    '0x7b79995e5f793A07Bc00c21412e50Ecae098E7f9',
    18,
    'WETH',
    'Wrapped Ether',
    'https://weth.io'
  ),
  // TODO: add correct contract
  [ChainId.BOB]: new ERC20Token(
    ChainId.BOB,
    '0x0000000000000000000000000000000000000000',
    18,
    'WETH',
    'Wrapped Ether',
    'https://weth.io'
  ),
  // TODO: add correct contract
  [ChainId.BOB_SEPOLIA]: new ERC20Token(
    ChainId.BOB_SEPOLIA,
    '0x4200000000000000000000000000000000000006',
    18,
    'WETH',
    'Wrapped Ether',
    'https://weth.io'
  )
};
