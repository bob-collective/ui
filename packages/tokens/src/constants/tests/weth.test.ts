import { describe, it, expect } from 'vitest';

import { WETH9 } from '../weth';

describe('WETH9 Snapshot', () => {
  it('should match snapshot for WETH9', () => {
    expect(WETH9).toMatchSnapshot();
  });
});
