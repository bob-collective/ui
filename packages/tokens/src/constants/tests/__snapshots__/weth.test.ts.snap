// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WETH9 Snapshot > should match snapshot for WETH9 1`] = `
{
  "1": ERC20Token {
    "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
    "chainId": 1,
    "decimals": 18,
    "isNative": false,
    "isToken": true,
    "name": "Wrapped Ether",
    "projectLink": "https://weth.io",
    "symbol": "WETH",
  },
  "11155111": ERC20Token {
    "address": "0x7b79995e5f793A07Bc00c21412e50Ecae098E7f9",
    "chainId": 11155111,
    "decimals": 18,
    "isNative": false,
    "isToken": true,
    "name": "Wrapped Ether",
    "projectLink": "https://weth.io",
    "symbol": "WETH",
  },
  "60808": ERC20Token {
    "address": "0x0000000000000000000000000000000000000000",
    "chainId": 60808,
    "decimals": 18,
    "isNative": false,
    "isToken": true,
    "name": "Wrapped Ether",
    "projectLink": "https://weth.io",
    "symbol": "WETH",
  },
  "808813": ERC20Token {
    "address": "0x4200000000000000000000000000000000000006",
    "chainId": 808813,
    "decimals": 18,
    "isNative": false,
    "isToken": true,
    "name": "Wrapped Ether",
    "projectLink": "https://weth.io",
    "symbol": "WETH",
  },
}
`;
